File: Data Structures & Algorithms/125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 31/05/2026, 12:02:59
class Solution {
public:
    bool isPalindrome(string s) {
        string res = "";
        for(auto key : s){
            if((key >= 'a' && key <= 'z') || (key >= 'A' && key <= 'Z') || (key >= '0' && key <= '9')){
                res.push_back(key);
            }
        }

        int low = 0;
        int high = res.size() - 1;
        while(low <= high){
            if(toupper(res[low]) != toupper(res[high])){
                return false;
            }
            low++;
            high--;
        }
        return true;
    }
};

File: Data Structures & Algorithms/125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 07/07/2026, 09:09:53
class Solution {
    public boolean isPalindrome(String s) {
       String res = "";
       for(int i=0;i<s.length();i++){
        if((s.charAt(i) >= 'a' && s.charAt(i) <= 'z') || (s.charAt(i) >= 'A' && s.charAt(i) <= 'Z') || (s.charAt(i) >= '0' && s.charAt(i) <= '9')){
            res = res + s.charAt(i);
        } 
       } 
       int i = 0;
       int j = res.length() - 1;
       while(i<=j){
        if(Character.toLowerCase(res.charAt(i)) != Character.toLowerCase(res.charAt(j))){
            return false;
        }
        i++;
        j--;
       }
       return true;
    }
}